src/libraries/wizard/metadatacomponentselector.cpp

```diff
@@ -0,0 +1,98 @@
+/****************************************************************************
+**
+** Copyright (C) 2017 N7 Mobile sp. z o. o.
+** Contact: http://n7mobile.pl/Space
+**
+** This file is part of ASN.1/ACN Plugin for QtCreator.
+**
+** Plugin was developed under a programme and funded by
+** European Space Agency.
+**
+** This Plugin is free software: you can redistribute it and/or modify
+** it under the terms of the GNU General Public License as published by
+** the Free Software Foundation, either version 3 of the License, or
+** (at your option) any later version.
+**
+** This Plugin is distributed in the hope that it will be useful,
+** but WITHOUT ANY WARRANTY; without even the implied warranty of
+** MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+** GNU General Public License for more details.
+**
+** You should have received a copy of the GNU General Public License
+** along with this program.  If not, see <http://www.gnu.org/licenses/>.
+**
+****************************************************************************/
+
+#include "metadatacomponentselector.h"
+
+#include <QDirIterator>
+
+#include <libraries/librarystorage.h>
+
+using namespace Asn1Acn::Internal::Libraries;
+using namespace Asn1Acn::Internal::Libraries::Wizard;
+
+MetadaComponentSelector::MetadaComponentSelector(const QString &path)
+    : m_path(path)
+{
+}
+
+QStringList MetadaComponentSelector::pathsFromNames(const QStringList &names)
+{
+    if (names.empty())
+        return {};
+
+    QDirIterator it(m_path, names, QDir::NoFilter, QDirIterator::Subdirectories);
+
+    QStringList paths;
+    while (it.hasNext())
+        paths.append(it.next());
+
+    return paths;
+}
+
+namespace {
+
+QStringList readElement(const Metadata::Element *element)
+{
+    if (!element->checked())
```

Comment:
logic that when something get checked something other might need to be checked too will be implemented in next tasks right? will it be separate object, or will it go into `setChecked` in Node?

Reply:
I think it will be `setChecked` from _Node_ to be used.

Reply:
ack